extension.js

```diff
@@ -173,6 +173,25 @@ const ExtensionLayout = new Lang.Class({
     GLib.spawn_command_line_async(cmd);
   },
 
+  _findNewStreamerEntries:function(lastList, currentList, detectGameChange) {
+    detectGameChange = detectGameChange || false;
+    if ( lastList.length == 0 )
+      return currentList;
+
+    let streamers = new Map();
+    lastList.forEach( entry => streamers.set(entry.streamer, entry) );
```

Comment:
This version breaks after a while, it seems. Looks like it happens when new streamers show up. Non-events and game changes don't seem to be an issue.